.travis.yml

```diff
@@ -77,11 +77,11 @@ before_install:
 
 install:
   - composer install --prefer-source --no-suggest
-  - |
-        if [ "${PHP_MAJOR}.${PHP_MINOR}" != "5.5" ]; then
-            composer remove --dev phpunit/phpunit
-            composer require -q -n --dev --no-progress --prefer-dist --no-suggest "phpunit/phpunit:5.7.*"
-        fi
+  # - |
+  #       if [ "${PHP_MAJOR}.${PHP_MINOR}" != "5.5" ]; then
+  #           composer remove --dev phpunit/phpunit
+  #           composer require -q -n --dev --no-progress --prefer-dist --no-suggest "phpunit/phpunit:5.7.*"
+  #       fi
```

Comment:
This change led to problem:

```
PHP Strict standards:  Declaration of Extension\Issue1404Test::onNotSuccessfulTest() should be compatible with PHPUnit_Framework_TestCase::onNotSuccessfulTest(Exception $e) in /home/travis/build/phalcon/zephir/unit-tests/Extension/Issue1404Test.php on line 250
PHP Stack trace:
PHP   1. {main}() /home/travis/build/phalcon/zephir/unit-tests/phpunit:0
PHP   2. PHPUnit_TextUI_Command::main() /home/travis/build/phalcon/zephir/unit-tests/phpunit:38
PHP   3. PHPUnit_TextUI_Command->run() /home/travis/build/phalcon/zephir/vendor/phpunit/phpunit/src/TextUI/Command.php:100
PHP   4. PHPUnit_Runner_BaseTestRunner->getTest() /home/travis/build/phalcon/zephir/vendor/phpunit/phpunit/src/TextUI/Command.php:123
PHP   5. PHPUnit_Framework_TestSuite->addTestFiles() /home/travis/build/phalcon/zephir/vendor/phpunit/phpunit/src/Runner/BaseTestRunner.php:58
PHP   6. PHPUnit_Framework_TestSuite->addTestFile() /home/travis/build/phalcon/zephir/vendor/phpunit/phpunit/src/Framework/TestSuite.php:384
PHP   7. PHPUnit_Util_Fileloader::checkAndLoad() /home/travis/build/phalcon/zephir/vendor/phpunit/phpunit/src/Framework/TestSuite.php:310
PHP   8. PHPUnit_Util_Fileloader::load() /home/travis/build/phalcon/zephir/vendor/phpunit/phpunit/src/Util/Fileloader.php:38
Strict standards: Declaration of Extension\Issue1404Test::onNotSuccessfulTest() should be compatible with PHPUnit_Framework_TestCase::onNotSuccessfulTest(Exception $e) in /home/travis/build/phalcon/zephir/unit-tests/Extension/Issue1404Test.php on line 250
Call Stack:
    0.0002          0   1. {main}() /home/travis/build/phalcon/zephir/unit-tests/phpunit:0
    0.0136          0   2. PHPUnit_TextUI_Command::main() /home/travis/build/phalcon/zephir/unit-tests/phpunit:38
    0.0137          0   3. PHPUnit_TextUI_Command->run() /home/travis/build/phalcon/zephir/vendor/phpunit/phpunit/src/TextUI/Command.php:100
    0.2155          0   4. PHPUnit_Runner_BaseTestRunner->getTest() /home/travis/build/phalcon/zephir/vendor/phpunit/phpunit/src/TextUI/Command.php:123
    0.2470          0   5. PHPUnit_Framework_TestSuite->addTestFiles() /home/travis/build/phalcon/zephir/vendor/phpunit/phpunit/src/Runner/BaseTestRunner.php:58
    0.7182          0   6. PHPUnit_Framework_TestSuite->addTestFile() /home/travis/build/phalcon/zephir/vendor/phpunit/phpunit/src/Framework/TestSuite.php:384
    0.7187          0   7. PHPUnit_Util_Fileloader::checkAndLoad() /home/travis/build/phalcon/zephir/vendor/phpunit/phpunit/src/Framework/TestSuite.php:310
    0.7188          0   8. PHPUnit_Util_Fileloader::load() /home/travis/build/phalcon/zephir/vendor/phpunit/phpunit/src/Util/Fileloader.php:38
```

Please take a look at https://travis-ci.org/phalcon/zephir/jobs/387499801

Reply:
Ugh. Ok. PHPUnit itself is inconsistent on this one.

I changed it because I was seeing the opposite error on my system (and in Travis), where the method in question does _not_ have the type hint, and PHP is complaining that the child class _does_. Since it's more correct for child classes to relax type restrictions than to be more strict about them (a change in 7.1 or 7.2, if I remember correctly, made this explicit by removing the error messages for that case), I think the revision is the more correct one, but we might be able to figure out another approach that lets the class support both parents, somehow...

Reply:
@danhunsaker Can you please try to sort out with this?

Reply:
I just realized I was responding to the error message rather than the actual changes being flagged. My above comment applies to a change in the `Issue1404Test` class itself, wherein I dropped the type hint. That, ultimately, is the actual cause of the error message above.

What's going on _here_ is that I moved the "use PHPUnit 5.7 if we're not using PHP 5.5" logic to Composer's realm of responsibility, by adjusting the version requirement in `composer.json` to allow `4.8` _or_ `5.7`, as appropriate. Since 5.7 doesn't support PHP 5.5, but does support 5.6 and 7.x, Composer automatically installs 4.8 for PHP 5.5, and 5.7 for all the others. No shell commands required.

I probably ought to remove those lines rather than comment them out, though.

Reply:
As to the inheritance error, I'll see what I can figure out.

Reply:
Hmmm. None of the approaches I can think of to conditionally define this method are working. I tried dropping it into a trait, and then defining that trait based on the PHP version, but PHPCS complained about putting the trait into the same file as the class, as well as putting it into a single file, since the keyword `trait` appeared twice, and only one trait/class/interface should be defined per file (which I agree with, but there didn't seem to be a way to tell PHPCS that both definitions were for the same trait; I'll happily switch back to that approach in a heartbeat if we have a workaround for that). It doesn't work at all to have a trait file for the autoloader that `require`s the correct actual trait file depending on PHP version, as the `require` statements seem to not be executed, so the trait is never actually defined before it gets used. (Though I might be able to get away with manually `require`ing the appropriate trait file _manually_ instead of relying on the autoloader, I suppose...) And attempting to override the method in `__construct`... just doesn't do anything.

Additional feedback is welcome on this one.

Reply:
Never mind. Went with the manual conditional `require_once()` option. This message should be gone.